backend/common/src/main/java/ai/verta/modeldb/common/CommonUtils.java

```diff
@@ -42,6 +43,23 @@ public static String appendOptionalTelepresencePath(String filePath) {
     }
   }
 
+  public static List<String> checkEntityTagsLength(List<String> tags) {
+    for (String tag : tags) {
+      if (tag.isEmpty()) {
+        var errorMessage = "Invalid tag found, Tag shouldn't be empty";
+        throw new ModelDBException(errorMessage, Code.INVALID_ARGUMENT);
+      } else if (tag.length() > CommonConstants.TAG_LENGTH) {
+        String errorMessage =
+            "Tag name can not be more than "
+                + CommonConstants.TAG_LENGTH
+                + " characters. Limit exceeded tag is: "
+                + tag;
+        throw new ModelDBException(errorMessage, Code.INVALID_ARGUMENT);
+      }
+    }
+    return tags;
+  }
```

Comment:
Let's move this to the tags handler since it seems tag specific. It's better to consolidate logic in groups vs spread around.

Reply:
Moved to common tag handler